packages/hdwallet-native/src/crypto/CryptoHelper.ts

```diff
@@ -148,4 +151,18 @@ export default class CryptoHelper {
     const entropy = await this.#engine.randomBytes(strength / 8);
     return entropyToMnemonic(Buffer.from(entropy));
   }
+
+  /**
+   * Generates a base64 hash based on the provided data
+   * Should be used to calculate a device ID for the wallet based on the mnemonic
+   */
+  async getDeviceId(data: string): Promise<string> {
+    if (!(typeof data === "string" && data.length > 0)) throw new Error("Invalid data to hash");
```

Comment:
The ! out front makes the logic harder for me to read. I know it doesnt really matter but  what about "dataType != string || data.length <= 0" or something?

Reply:
I do it like that because it's easier for me to think of the cases where the validation passes than when it doesn't, so I write the good cases then wrap that in the `!` to negate it, but I do understand where you're coming from.
I'd prefer to use a standard validation process (library or home grown) to make everything consistent.

Reply:
Do you need the `typeof data === "string"` check if the function signature requires data to be a string?

Reply:
TypeScript only enforces types at compile time not at run time. This is an open source library and anyone could use it with plain JavaScript. There's nothing in that scenario that will prevent them from passing invalid data. I prefer that libraries I use throw errors when I use them incorrectly rather than just do unexpected things that are hard to debug.